GitCommands/Settings/ConfigFileSettingsCache.cs

```diff
@@ -89,14 +106,64 @@ public IList<ConfigSection> GetConfigSections()
             });
         }
 
-        public void RemoveConfigSection(string configSectionName)
+        /// <summary>
+        /// Removes the specific configuration section from the .git/config file.
+        /// </summary>
+        /// <param name="configSectionName">The name of the configuration section.</param>
+        /// <param name="performSave">If <see langword="true"/> the configuration changes will be saved immediately.</param>
+        public void RemoveConfigSection(string configSectionName, bool performSave = false)
         {
             LockedAction(() =>
             {
                 EnsureSettingsAreUpToDate();
                 _configFile.Value.RemoveConfigSection(configSectionName);
+                if (performSave)
+                {
+                    _configFile.Value.Save();
+                }
             });
         }
 
+        /// <summary>
+        ///  Marks the remote as enabled or disabled in .git/config file.
+        /// </summary>
+        /// <param name="remoteName">An existing remote instance.</param>
+        /// <param name="disabled">The new state of the remote. <see langword="true"/> to disable the remote; otherwise <see langword="false"/>.</param>
+        /// <param name="onRemoteDisabled">Action to perform whenever the remote is disabled.</param>
+        public void ToggleRemoteState(string remoteName, bool disabled, Action<string> onRemoteDisabled)
+        {
+            if (string.IsNullOrWhiteSpace(remoteName))
+            {
+                throw new ArgumentException(nameof(remoteName));
+            }
+            if (onRemoteDisabled == null)
+            {
+                throw new ArgumentNullException(nameof(onRemoteDisabled));
+            }
+
+            LockedAction(() =>
+            {
+                // disabled is the new state, so if the new state is 'false' (=enabled), then the existing state is 'true' (=disabled, i.e. '-remote')
+                var sectionName = (disabled ? "" : ConfigFile.DisabledSectionPrefix) + ConfigFile.SectionRemote;
+
+                var sections = GetConfigSections();
+                var section = sections.FirstOrDefault(s => s.SectionName == sectionName && s.SubSection == remoteName);
+                if (section == null)
+                {
+                    // we didn't find it, nothing we can do
+                    return;
+                }
+
+                // rename the remote
+                section.SectionName = (disabled ? ConfigFile.DisabledSectionPrefix : "") + ConfigFile.SectionRemote;
+
+                if (disabled)
+                {
+                    // execute 'git remote remove' command - it will clean up
```

Comment:
You also will end up with such a state without calling `git remote remove`.
And you can still see branches from the disabled remote (for me it is an advantage).
Removing remote you will not see those branches when remote is enabled again (until you fetch them again).